docs/quickstart.rst

```diff
@@ -8,51 +8,156 @@ up pg_auto_failover to replicate data between them. We’ll simulate failure in
 the primary node and see how the system smoothly switches (fails over)
 to the secondary.
 
-For simplicity we’ll run all the pieces on a single machine, but in
-production there would be three independent machines involved for each
-managed PostgreSQL service. One machine for the primary, another for
-the secondary, and the last as a monitor which watches the other nodes’
-health, manages global state, and assigns nodes their roles.
+For illustration, we'll run our databases on virtual machines in the Azure
+platform, but the techniques here are relevant to any cloud provider or
+on-premise network. We'll use four virtual machines: a primary database, a
+secondary database, a monitor, and an "application." The monitor watches the
+other nodes’ health, manages global state, and assigns nodes their roles.
+
+.. _quickstart_network:
+
+Create virtual network
+----------------------
+
+Our database machines need to talk to each other and to the monitor node, so
+let's create a virtual network.
+
+.. code-block:: bash
+
+   az group create \
+       --name ha-demo \
+       --location eastus
+
+   az network vnet create \
+       --resource-group ha-demo \
+       --name ha-demo-net \
+       --address-prefix 10.0.0.0/16
+
+We need to open ports 5432 (Postgres) and 22 (SSH) between the machines, and
+also give ourselves address from our remote IP. We'll do this with a network
+security group and a subnet.
+
+.. code-block:: bash
+
+   az network nsg create \
+       --resource-group ha-demo \
+       --name ha-demo-nsg
+
+   az network nsg rule create \
+       --resource-group ha-demo \
+       --nsg-name ha-demo-nsg \
+       --name ha-demo-ssh-and-pg \
+       --access allow \
+       --protocol Tcp \
+       --direction Inbound \
+       --priority 100 \
+       --source-address-prefixes `curl ifconfig.me` 10.0.1.0/24 \
+       --source-port-range "*" \
+       --destination-address-prefix "*" \
+       --destination-port-ranges 22 5432
+
+   az network vnet subnet create \
+       --resource-group ha-demo \
+       --vnet-name ha-demo-net \
+       --name ha-demo-subnet \
+       --address-prefixes 10.0.1.0/24 \
+       --network-security-group ha-demo-nsg
+
+Finally add four virtual machines (ha-demo-a, ha-demo-b, ha-demo-monitor, and
+ha-demo-app). For speed we background the ``az vm create`` processes and run
+them in parallel:
+
+.. code-block:: bash
+
+   # allow VMs to resolve each other via internal DNS
+   az network private-dns zone create \
+       --resource-group ha-demo \
+       --name ha-demo.local
+   az network private-dns link vnet create \
+       --name ha-demo-zone-link \
+       --resource-group ha-demo \
+       --virtual-network ha-demo-net \
+       --zone-name ha-demo.local \
+       --registration-enabled true
```

Comment:
I think this should be another separate box before we create the VMs. It makes tracking where we are at easier, and also makes more sense (to me at least) logically.